### PR TITLE
don't ignore errors that occur when the TLS ClientHello is generated

### LINKED ISSUE #3419: not every crypto/tls error generates an alert
Errors that occur during `makeClientHello` don't generate an alert, and are currently silently ignored by the crypto setup: https://github.com/lucas-clemente/quic-go/blob/60bbe92194a5d4d90bfb3f142cbab69b9cb51561/internal/handshake/crypto_setup.go#L302-L323

We should return the error the caller instead.

This bug is easy to reproduce by not setting an empty string in `qtls.Config.NextProtos`, but there are numerous other error paths.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #3419.

### CODE REVIEW

File: internal/handshake/crypto_setup.go [hunk 1]
```diff
@@ -113,7 +113,8 @@ type cryptoSetup struct {
 
 	zeroRTTParameters      *wire.TransportParameters
 	clientHelloWritten     bool
-	clientHelloWrittenChan chan *wire.TransportParameters
+	clientHelloWrittenChan chan struct{} // is closed as soon as the ClientHello is written
```
Comment (xieyuschen) [inline internal/handshake/crypto_setup.go:116]: Is there some reasons to change the type from `chan *wire.TransportParameters` from `chan struct{}`?

Comment (marten-seemann) [reply internal/handshake/crypto_setup.go:116]: It was not changed. The channel was renamed to `zeroRTTParametersChan`, and a new `clientHelloWrittenChan` was added.

Comment (xieyuschen) [reply internal/handshake/crypto_setup.go:116]: > It was not changed. The channel was renamed to `zeroRTTParametersChan`, and a new `clientHelloWrittenChan` was added.

Ohh, thanks for pointing it out.

**Review by lucas-clemente — APPROVED** [review]

**Comment by marten-seemann** [comment]
Wrong merge method. Will revert and re-merge.

**Pull request merged**